### PR TITLE
[TensorDec] Add data struct for tensorflow lite labels - @open sesame 09/13 10:19

### PR DESCRIPTION
**Pull request by jinhyuck-park**
Add initial data struct to process labels for tensor from tensor filter

Signed-off-by: jinhyuck-park <jinhyuck83.park@samsung.com>

**Self evaluation:**
1. Build test: [* ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [* ]Skipped

### CODE REVIEW

File: gst/tensor_decoder/tensordec.c [hunk 1]
```diff
@@ -96,9 +97,30 @@ enum
...
 
+/**
+ * @brief Data structure for tflite model info.
+ */
+typedef struct
+{
+  gchar *label_path; /**< label file path */
+  GList *labels; /**< list of loaded labels */
+  guint total_labels; /**< count of labels */
+} tflite_info_s;
```
Comment (myungjoo) [inline gst/tensor_decoder/tensordec.c]: The data structure looks good

File: gst/tensor_decoder/tensordec.c [hunk 2]
```diff
@@ -96,9 +97,30 @@ enum
...
+
+/**
+ * @brief Data structure for tensor decoder.
+ */
+typedef struct
+{
+  gint current_label_index; /**< current label index */
+  gint new_label_index; /**< new label index */
+  tflite_info_s tflite_info; /**< tflite model info */
+} TensorDec_Data;
```
Comment (myungjoo) [inline gst/tensor_decoder/tensordec.c]: tensordec should NOT have dependency on tflite.

File: gst/tensor_decoder/tensordec.c [hunk 3]
```diff
@@ -96,9 +97,30 @@ enum
 {
   PROP_0,
   PROP_OUTPUT_TYPE,
-  PROP_SILENT
+  PROP_SILENT,
+  PROP_LABEL
```
Comment (myungjoo) [inline gst/tensor_decoder/tensordec.c]: Labeling is only one of the possible options. Thus, I'd suggest:
```
PROP_MODE,
PROP_MODE_OPTION1,
PROP_MODE_OPTION2,
```

Then, ```MODE``` = ```{ "direct_video", "image_labeling" }``` at this version.
With ```direct_video``` mode, ```MODE_OPTION*``` is ignored.
With ```image_labeling" mode, ```MODE_OPTION1``` is the file path to the label file.

THen, with the same property variables, we can support other "modes" later without adding more property variables.

**Comment by myungjoo** [comment]
https://github.com/nnsuite/nnstreamer/tree/master/gst/tensor_decoder has some suggestions. But when you design PROPERTIES, please consider:

1. We do not want to add PROPERTIES with additional decoding mechanisms. We want to reuse.
2. Some will require files such as label files. Some will require other options such as colors or fonts. Some will require both (bounding box colors and label file)
3. Some may want multiple files.

You need to supply something that can support all although you are supporting image-label with this sprint. Further modifications in PROPERTIES should be minimized if not removed entirely.

That gst/tensor_decoder/README.md is still not complete, please update as you implement your code.

**Comment by leemgs** [comment]
@myungjoo We need your help. :) Could you run "**bot.net$ sudo pbuild create**" command to support the ubuntu package build test at the nnsuite.mooo.com server?

**Comment by leemgs** [comment]
In case that you have experiences "pr-audit-build-ubuntu" issue, Please refer to the below issue #241.
https://github.com/nnsuite/TAOS-CI/issues/241

**Comment by myungjoo** [comment]
Ignoring Ubuntu BUild CI errors as it is failing.

**Pull request merged**